src/NuGet.Core/NuGet.PackageManagement/Projects/NuGetProject.cs

```diff
@@ -142,5 +153,22 @@ public static string GetUniqueNameOrName(NuGetProject nuGetProject)
 
             return nuGetProjectName;
         }
+
+        public static async Task<string> GetUniqueNameOrNameAsync(NuGetProject nuGetProject, CancellationToken token)
```

Comment:
`UniqueName` is a Visual Studio/solution specific concept, so I'm not a huge fan of adding it into this lower level API that is also used on command line and 3rd party apps.

Our VS code has something called a ProjectAdapter. It might already have methods to get unique name. Did you investigate this class, and how to get a ProjectAdapter from a NuGetProject from within VS?

I'm scared that it might have a cascading effect and require lots of changes, but could this method be moved to `NuGetProjectInternal`, and either cast to that type in the VS code, or change the classes that hold references of `NuGetProject` to use `NuGetProjectInternal` instead?

Reply:
This is just an Async version of a synchronous method already exists in this class, I am not adding new functionality or a new concept here.